### ISSUE TITLE
Weird render

### ISSUE THREAD

**Issue by lotziko** [Relfos/LunarFonts#1]
Hello! I cloned the repo and tried to launch the demo but got that result in output.png:

![image](https://user-images.githubusercontent.com/38538633/51637950-c348d900-1f65-11e9-9955-346d6a1f359c.png)

Also i tried nuget package which i added to my OpenTK project and got the same result.

**Comment by Relfos** [comment]
Hi, what TTF font are you using?
I've tested with around 20 fonts, most worked fine, I find a couple of fonts more complex that crashed the demo, but I've not yet found one that caused something like that.

**Comment by lotziko** [comment]
Tried it with arial.ttf and segoeui.ttf.

**Comment by lotziko** [comment]
I tried some more fonts and got results:

Not works: Calibri, Times New Roman, Segoe Script, Microsoft Sans Serif Regular, Tahoma.
Works: Candara, Comic Sans, Consolas, Cambria.

What about original C code? Is it working with every font?

**Comment by Relfos** [comment]
I've added source code for a font viewer that allows to quickly test all fonts inside a folder.
I now used this to quickly take a look at the folders installed in my computer and really the more "important" fonts like Arial, Tahoma, Segoeu and others do not work, everything else mostly works.

I will now check the original C version, if it does work, then it means I introduced a small bug during the port to C#, if that's the case, I will try fix it.

**Comment by Relfos** [comment]
Hi, no updates yet, as I no longer use this lib myself.
If having this work is important to you, the original C code is here:
https://github.com/nothings/stb/blob/master/stb_truetype.h

I'm sorry but right now I'm swamped with other work so I cant look at it. If you have Visual Studio installed you can try creating a C project there, that lib is a single file so its easy to compile. Then load a ttf font there and do a step by step compare of the variables with the C# version until you find a place where the values dont match, and once you find that, should be easy to fix the C# version.  Its possible that its a very tiny change, even a single line of code or two that you might need to change

**Comment by GoldenretriverYT** [comment]
Thanks for the reply, but that probably wont really be possible because even the oldest version of stb_truetype contains stuff this library doesnt even contain (BakedChar, etc.), so there really isnt a good way to compare anything. And I believe, that the missing 3k lines since 2014 are probably the reason.

But I understand that you have no time and no use of updating this library, so I will just try to port the newest version myself

**Comment by Relfos** [comment]
Yes, now that you say that, this library was originally ported around 9 years ago or so, and it was C to Delphi, then later Delphi to C#, so I guess it's no longer similar at all to the latest stb_truetype. I guess this bug never was fixed here because the fonts that I used myself worked fine.
But if you have the motivation for porting the latest version, its a small lib, should take a few days max.
